R/zchunk_L203.water.mapping.R

```diff
@@ -38,103 +37,134 @@ module_water_L203.water.mapping_DISABLED <- function(command, ...) {
-    # ===================================================
 
-    # Produce outputs
-    # Temporary code below sends back empty data frames marked "don't test"
-    # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
-      add_legacy_name("L203.SectorLogitTables[[ curr_table ]]$data") %>%
-      add_precursors("precursor1", "precursor2", "etc") %>%
-      # typical flags, but there are others--see `constants.R`
-      add_flags(FLAG_LONG_YEAR_FORM, FLAG_NO_XYEAR) ->
-      L203.SectorLogitTables[[ curr_table ]]$data
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    # Create tibble with all possible mapping sectors...
+
+    # (a) irrigation sectors
+    L125.LC_bm2_R_GLU %>% ungroup %>%
+      select(GCAM_region_ID, GLU) %>%
+      left_join_error_no_match(select(basin_to_country_mapping, GLU_code, GLU_name), by = c("GLU" = "GLU_code")) %>%
+      # ^^ join GLU names, which will replace GLU codes in this tibble
+      select(-GLU) %>% rename(GLU = GLU_name) %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      repeat_add_columns(filter(A03.sector, water.sector %in% IRRIGATION)) %>%
+      repeat_add_columns(tibble(water_type = c("water consumption", "water withdrawals"))) %>%
+      mutate(supplysector = set_water_input_name(water.sector, water_type, A03.sector, GLU)) ->
+      L203.mapping_irr
+
+    # (b) non-irrigation sectors
+    GCAM_region_names %>%
+      repeat_add_columns(filter(A03.sector, !(water.sector %in% IRRIGATION))) %>%
+      mutate(GLU = NA) %>%
+      repeat_add_columns(tibble(water_type = c("water consumption", "water withdrawals"))) %>%
+      mutate(supplysector = set_water_input_name(water.sector, water_type, A03.sector)) ->
+      L203.mapping_nonirr
+
+    # (c) combine irrigation and non-irrigation sectors and add additional required columns
+    L203.mapping_irr %>%
+      bind_rows(L203.mapping_nonirr) %>%
+      mutate(coefficient = 1,
+             subsector = supplysector,
+             technology = supplysector,
+             logit.year.fillout = first(BASE_YEARS)) %>%
+      arrange(GCAM_region_ID) %>%
+      left_join(select(L165.ag_IrrEff_R, -field.eff), by = "GCAM_region_ID") %>%
+      # ^^ non-restrictive join required (NA values generated for region 30, Taiwan)
+      mutate(coefficient = if_else(water.sector == IRRIGATION & water_type == "water withdrawals",
```

Comment:
Please put in a comment that the conveyance loss is taken out of agricultural withdrawals and placed in as a loss in the water distribution sectors (water_td_irr). Thus, to get comparable total agriculture withdrawals for reporting, the conveyance loss must be included.  This is a database query issue, but I don't where else we can document it.  Also originally, I had used efficiency (conveyance.eff) directly rather than 1/eff, which makes it easier to interpret the conveyance loss since its a fraction less than 1, rather than a coefficient greater than 1. Please put in a comment that maybe we should reconsider that.